code/modules/power/proto_sh_emitter.dm

```diff
@@ -0,0 +1,293 @@
+///Amount of drained power per tile
+#define TILE_POWER_CONSUMPTION 3500
+
+/obj/machinery/power/proto_sh_emitter
+	name = "Prototype Shield Emitter"
+	desc = "This is a Prototype Shield Emitter that create in front of it a box made of shielding elements to protect the station from heat and pressure"
+	icon = 'icons/obj/power.dmi'
+	icon_state = "proto_sh_emitter"
+	anchored = FALSE
+	density = TRUE
+	max_integrity = 350
+	integrity_failure = 0.2
+	circuit = /obj/item/circuitboard/machine/proto_sh_emitter
+	///Store the powered shields placed in the world, used when turned off to removed them
+	var/list/signs
+	///Check if the machine is turned on or off
+	var/is_on = FALSE
+	///Used to check if the machine is placed inside the borders of the map
+	var/borders = TRUE
+	///Is the machines currently projecting a barrier?
+	var/has_barrier = FALSE
+	///Vars used in the GUI to be able to setup a size
+	var/a_i = 1
+	var/b_i = 1
+	var/c_i = 2
+	var/a_o = 2
+	var/b_o = 2
+	var/c_o = 3
+
+/obj/machinery/power/proto_sh_emitter/anchored
+	anchored = TRUE
+
+/obj/machinery/power/proto_sh_emitter/Initialize()
+	. = ..()
+	var/area/current_area = get_area(src)
+	if(!current_area)
+		return
+	RegisterSignal(current_area, COMSIG_AREA_POWER_CHANGE, .proc/check_power)
+
+/obj/machinery/power/proto_sh_emitter/Destroy()
+	if(SSticker.IsRoundInProgress())
+		var/turf/T = get_turf(src)
+		stack_trace("Prototype Shield Emitter deleted at [ADMIN_VERBOSEJMP(T)]")
+		log_game("Prototype Shield Emitter deleted at [AREACOORD(T)]")
+	QDEL_LIST(signs)
+	return ..()
+
+/obj/machinery/power/proto_sh_emitter/update_icon_state()
+	. = ..()
+	if(has_barrier)
+		icon_state = "proto_sh_emitter_on"
+	else
+		icon_state = "proto_sh_emitter"
+
+/obj/machinery/power/proto_sh_emitter/attackby(obj/item/I, mob/user, params)
+	if(!is_on)
+		if(default_deconstruction_screwdriver(user, "proto_sh_emitter_open", "proto_sh_emitter", I))
+			return
+	if(default_change_direction_wrench(user, I))
+		return
+	if(default_deconstruction_crowbar(I))
+		return
+	return ..()
+
+/obj/machinery/power/proto_sh_emitter/attackby_secondary(obj/item/item, mob/user, params)
+	. = ..()
+	if(panel_open && item.tool_behaviour == TOOL_WRENCH)
+		if(default_unfasten_wrench(user, item))
+			return SECONDARY_ATTACK_CONTINUE_CHAIN
+
+/obj/machinery/power/proto_sh_emitter/process()
+	if(!is_on)
+		return
+	var/power_use = idle_power_usage
+	if(LAZYLEN(signs))
+		for(var/i in 1 to LAZYLEN(signs))
+			power_use += TILE_POWER_CONSUMPTION
+	use_power(power_use)
```

Comment:
Does it actually need to process like this? could we update the idle_power_usage when signs are added/lost?